clients/src/main/java/org/apache/kafka/clients/admin/FeatureUpdate.java

```diff
@@ -23,33 +23,92 @@
  */
 public class FeatureUpdate {
     private final short maxVersionLevel;
-    private final boolean allowDowngrade;
+    private final DowngradeType downgradeType;
+
+    public enum DowngradeType {
```

Comment:
It's kind of awkward to upgrade by setting DowngradeType = NONE. Can we have this be UpdateType and have UPGRADE, DOWNGRADE, UNSAFE_DOWNGRADE?